tuf/signed/verify.go

```diff
@@ -77,10 +77,11 @@ func VerifySignatures(s *data.Signed, roleData data.BaseRole) error {
 		if key.ID() != sig.KeyID {
 			return ErrInvalidKeyID{}
 		}
-		if err := VerifySignature(msg, sig, key); err != nil {
+		if err := VerifySignature(msg, &sig, key); err != nil {
 			logrus.Debugf("continuing b/c %s", err.Error())
 			continue
 		}
+		s.Signatures[i].IsValid = sig.IsValid
```

Comment:
The double-assign between `VerifySignature` and `VerifySignatures` we have to do here for `IsValid` on the same signature feels like we should be using pointers to refer to `Signatures` from `Signed`, and only having `VerifySignature` do any mutation.

That said, it's a significant change for all of our data structures given that we aren't making heavy use of `IsValid`.  cc @endophage @cyli for more discussion - maybe instead we could have a separate utility function to filter valid signatures?